### DISCUSSION TITLE
What's set in zustand?

### DISCUSSION THREAD

**Discussion by akramnarejo** [pmndrs/zustand#871] [Q&A]
```
const useStore = create(set => ({
  bears: 0,
  increasePopulation: () => set(state => ({ bears: state.bears + 1 })),
  removeAllBears: () => set({ bears: 0 })
}))
```
In above code I want to understand, what's this **set** being passed into callback in create(set => ({})) function? and definition of property is wrapped with that **set** value.

**Comment by dai-shi** [comment]
While it's not explicitly written in docs (as it's sort of implementation details), `set` is equal to `useStore.setState`.
https://github.com/pmndrs/zustand/blob/da781a25778a36449135050d7f46c0e80bfefbcc/src/vanilla.ts#L98-L112

So technically, it's just a shorthand of this:
```js
const useStore = create(() => ({
  bears: 0,
  increasePopulation: () => useStore.setState(state => ({ bears: state.bears + 1 })),
  removeAllBears: () => useStore.setState({ bears: 0 })
}))
```

`useStore.setState` takes `partial` and optional `replace`. `partial` can be a partial state or a function that returns it (taking the previous state).

**Reply by raarts** [reply]
What does set() return though?

**Reply by dai-shi** [reply]
It doesn't return anything, `void`.
